app/src/hrValidation/java/org/sagebase/crf/HrMainActivity.java

```diff
@@ -0,0 +1,112 @@
+package org.sagebase.crf;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.MenuItem;
+import android.view.View;
+
+import org.researchstack.backbone.DataProvider;
+import org.researchstack.backbone.StorageAccess;
+import org.researchstack.backbone.factory.IntentFactory;
+import org.researchstack.backbone.result.TaskResult;
+import org.researchstack.backbone.task.Task;
+import org.researchstack.backbone.ui.MainActivity;
+import org.researchstack.backbone.ui.ViewTaskActivity;
+import org.researchstack.backbone.utils.LogExt;
+import org.sagebase.crf.view.CrfFilterableActivityDisplay;
+import org.sagebionetworks.bridge.researchstack.CrfResourceManager;
+import org.sagebionetworks.bridge.researchstack.CrfTaskFactory;
+import org.sagebionetworks.research.crf.R;
+
+import static org.researchstack.backbone.ui.fragment.ActivitiesFragment.REQUEST_TASK;
+
+public class HrMainActivity extends MainActivity {
+
+    private static final String LOG_TAG = HrMainActivity.class.getCanonicalName();
+
+    private CrfTaskFactory taskFactory = new CrfTaskFactory();
+    private IntentFactory intentFactory = IntentFactory.INSTANCE;
+
+    /** Intent factory, made available for subclasses to create Intent instances. */
+    public final IntentFactory getIntentFactory() {
+        return intentFactory;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        findViewById(R.id.toolbar).setVisibility(View.GONE);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (!DataProvider.getInstance().isSignedIn(this)) {
```

Comment:
Do `DataProvider.getInstance()` and `ResearchStack.getInstance()` work? I'm not seeing where they're initialized

Reply:
They get initialized in MainApplication.